pkg/providers/instancetype/suite_test.go

```diff
@@ -168,6 +168,16 @@ var _ = Describe("InstanceType Provider", func() {
 				ContainSubstring("kubernetes.azure.com/podnetwork-type=overlay"),
 			))
 		})
+		It("should use the subnet specified in the nodeclass", func() {
+			nodeClass.Spec.VNETSubnetID = lo.ToPtr("/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/sillygeese/providers/Microsoft.Network/virtualNetworks/karpenter/subnets/nodeclassSubnet")
```

Comment:
nit: make this string a const, at least in this function just to simplify that the comparison later is directly with the same string?